org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java

```diff
@@ -343,7 +345,22 @@ public void onComplete() {
                                     Integer.toString(mNewDevicesArrayAdapter.getCount())));
                         }
 
-                        showSnackbar("Discovery Finished!");
+                        // Set timer for 15sec
+                        new CountDownTimer(15000, 1000) {
+                            @Override
+                            public void onTick(long millisUntilFinished) {
+                            }
+                            @Override
+                            public void onFinish() {
+                                // If discovering of a device takes more than 15 sec ,then show user discovery is finished
+                                // if the needed device is not found , rediscover button can be used.
+                                if (mBluetoothHandler.isBluetoothEnabled()) {
+                                    mProgressBar.setVisibility(View.GONE);
+                                    mRescanImageView.setVisibility(View.VISIBLE);
+                                    showSnackbar("Discovery Finished!");
+                                }
+                            }
+                        }.start();
```

Comment:
Placing this into `onComplete()` does not has the desired effect. The `CountDownTimer` will only start after discovery has finished. So, why should we wait another 15 seconds for discovery? Wouldn't it be a better solution, to start the timer directly after clicking on the refresh button?

Reply:
Yes. I have placed this in `onStart()` and tested the functionality. Works fine in my device. Thank you testing this feature closely :)